src/common/src/system_param/adaptive_parallelism_strategy.rs

```diff
@@ -22,7 +22,10 @@ use risingwave_common::system_param::ParamValue;
 use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
-#[derive(PartialEq, Copy, Clone, Debug, Serialize, Deserialize, Default)]
+/// Use `#[serde(try_from, into)]` to serialize/deserialize as string format (e.g., "Bounded(64)"),
+/// which is consistent with `ALTER SYSTEM SET` command.
+#[derive(PartialEq, Copy, Clone, Debug, Default, Serialize, Deserialize)]
+#[serde(try_from = "String", into = "String")]
```

Comment:
Ideally we can use [`DeserializeFromStr`](https://docs.rs/serde_with/latest/serde_with/derive.DeserializeFromStr.html) to reuse the `parse` logic.